libraries/resource_providers/helpers/validations.rb

```diff
@@ -0,0 +1,28 @@
+# (c) Copyright 2017 Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed
+# under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+# CONDITIONS OF ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+module OneviewCookbook
+  # module for validations of resource providers
+  module Validations
+    # module with method related to validate the presence of properties
+    module Presence
+      # Validates the presence of properties
+      # @param [Symbol] property An property name to be validating presence
+      # @param [Symbol] ... More property names
+      # @raise [RuntimeError] if some property is not set
+      def validate_presence_of(*properties)
```

Comment:
It would be good to create a unit test. Because this method can be used by other resources, it can be a share example. The exception is not being tested.

Reply:
The unit tests of individual resources are testing this functionallity, because if presence_of properties is not set, the unit test will brake, raising an error.

Reply:
But, thinking abou this, I thought one enhacement, I can put this module included into resource_provider, all resources can use this funcionallity. I can to do test to expect exception when property is not present.
What do you think?

Reply:
@aalexmonteiro

Reply:
Aye, this likely makes more sense inside resource_provider than in helpers. 👍 for moving it.

Reply:
@fgbulsoni I'm thinking to move that method `validates_presence_of` to `oneview_helper.rb` (this Helper module have some auxiliar methods used by resource_provider, like `load_resource`), I guess make sense, I can put just the calling to the method into `resource_provider` class. With this I avoid have that change the class length into `.rubocop.yml`, because if that new method `validates_presence_of` be entire into `resource_provider`, `resource_provider` will have more than 200 lines.

Reply:
Or maybe, change the class length is good idea, because is less complex for one simples cause and common to all resource providers.

Reply:
I'd don't really see us calling this outside of this scenario, so I'm not really convinced about putting it in `oneview_helper`. I'd vote to just increase the length on rubocop and go with `resource_provider`

Reply:
It was not necessary increase the length. \o/ =D